src/pages/tags/[id].astro

```diff
@@ -32,17 +32,22 @@ const filteredPosts = await getCollection("blog", (p: Blog) => {
 const posts = filteredPosts.sort(
   (a, b) => b.data.pubDate.valueOf() - a.data.pubDate.valueOf(),
 );
+const title = `Posts tagged "${tagName}"`;
+const description =
+  posts.length > 0
+    ? `Nick has written ${posts.length} posts about this topic, which you can find here!`
+    : `Nick hasn't written any posts about this topic. Come back later!`;
 ---
 
 <!doctype html>
 <html lang="en">
   <head>
-    <BaseHead title={SITE_TITLE} description={SITE_DESCRIPTION} />
+    <BaseHead {title} {description} />
   </head>
   <body>
     <Header />
     <main id="content-start">
-      <h1>{tagName}</h1>
+      <h1>Posts tagged <code>{tagName}</code></h1>
       <Cards {posts} />
```

Comment:
Add a condition to show the description if there aren't any posts?

Reply:
This is not necessary, since Astro will only build pages for tags that exist. The condition will only happen if I create a tag file, but don't include the tag in any blogs.

For now, I'll ignore the condition, but I'll make a note of it if I want to create a more user friendly experience.

Reply:
Meh I just did it, so it's fine.